nemo_text_processing/g2p/modules.py

```diff
@@ -38,6 +38,13 @@
 from nemo.utils.decorators import experimental
 from nemo.utils.get_rank import is_global_rank_zero
 
+try:
+    from nemo_text_processing.g2p.models.heteronym_classification import HeteronymClassificationModel
```

Comment:
this library is only needed when `heteronym_model` in `setup_heteronym_model` is specified. Is it better to import it as needed during instanciating by calling `__init__`? something similar to https://github.com/NVIDIA/NeMo/blob/main/nemo_text_processing/g2p/modules.py#L702-L717.

```python
class BaseG2p(ABC):
    def __init__(
        self,
        phoneme_dict=None,
        word_tokenize_func=lambda x: x,
        apply_to_oov_word=None,
        mapping_file: Optional[str] = None,
        heteronym_model: Optional[str] = None,
    ):
        self.phoneme_dict = phoneme_dict
        self.word_tokenize_func = word_tokenize_func
        self.apply_to_oov_word = apply_to_oov_word
        self.mapping_file = mapping_file

        if heteronym_model is not None:
            try:
                from nemo_text_processing.g2p.models.heteronym_classification import HeteronymClassificationModel
            except ImportError as e:
                logging.error(e)

        self.heteronym_model = heteronym_model
```

Reply:
I moved this import to setup_heteronym_model(). I don't think this should be in init() as we discussed that during TTS training heteronyms should remain in grapheme. So that TTS can handle heteronyms not covered by the heteronym_classification model.